devnet/substrate_alice.Dockerfile

```diff
@@ -0,0 +1,40 @@
+FROM parity/polkadot:v0.9.10 AS polkadot
+FROM golang:1.17
+
+ARG CHAIN=cross-client
```

Comment:
what is the difference between the `gssmr` chain that's within our codebase and this one?

Reply:
Both have 3 authority nodes, however, the `cross-client` genesis file is accepted by the substrate node since it has `staking` and `session` json keys while `gssmr` chain has `Babe` and `Grandpa` json keys. The `cross-client` genesis file has the runtime version 09.10 that don't trigger errors like:
````
CRIT target=runtime message=panicked at 'Timestamp slot must match `CurrentSlot`'
````

Reply:
> where did the original genesis-spec.json come from?

`./polkadot build-spec --disable-default-bootnode --dev > genesis-spec.json`
And then you will have to modify it according to your requirements, like adding authorities. 

And then you run something like this to get a genesis.json file
`./polkadot build-spec --chain genesis-spec.json --raw --disable-default-bootnode > genesis.json`

Will make a pr to add this to read me.

New versions have a different genesis-spec file than 0.9.10.
Also running cross-client dev net with newer version of polkadot gives runtime errors. So, yeah we need to update.

Reply:
can we rename this to something more descriptive.  `3-auth-node-0.9.10` or something.

Reply:
Also we could have `0.9.10` as a build argument at the top global scope in the Dockerfile.

i.e.

```Dockerfile
ARG POLKADOT_VERSION=v0.9.10

FROM parity/polkadot:${POLKADOT_VERSION} AS polkadot
FROM golang:1.17

ARG CHAIN=3-auth-node-${POLKADOT_VERSION}
```

Reply:
@qdm12 I was having problems trying to use arguments declared before `FROM`, following this issue ([link](https://github.com/moby/moby/issues/34129)) I could resolve and apply the suggestion. 

Basically, all the `ARG`'s declared/initialized before `FROM` needs to be redeclared after `FROM` to use its values otherwise will not be possible to get the value from `ARG`. I your suggestion we need to add

```
ARG POLKADOT_VERSION=v0.9.10

FROM parity/polkadot:${POLKADOT_VERSION} AS polkadot
FROM golang:1.17

+ ARG POLKADOT_VERSION
# Now its possible to use the value here
ARG CHAIN=3-auth-node-${POLKADOT_VERSION} 
```